File: code/products/m3db/write.py
from datetime import datetime

from influxdb import InfluxDBClient


def main():
    client = InfluxDBClient(host="SERVICE_HOST",
                            port="SERVICE_PORT",
                            username="avnadmin",
                            password="AVNADMIN_PASS",
                            database="default",
                            ssl=True,
                            verify_ssl=True,
                            path='/api/v1/influxdb')

    json_body = [
        {
            "measurement": "cpu_load_short",
            "tags": {
                "host": "testnode",
            },
            "time": datetime.utcnow().isoformat(),
            "fields": {
                "value": 0.96
            }
        }
    ]
    print(client.write_points(json_body))


if __name__ == "__main__":
    main()

File: code/products/m3db/write.php
<?php

require "vendor/autoload.php";

// create a client
$client = new InfluxDB\Client(
    SERVICE_HOST,
    SERVICE_PORT,
    "avnadmin",
    AVNADMIN_PASS,
    true,
    true,
);

// construct a data point
$point = new InfluxDB\Point(
		'php_example_metric', // name of the measurement
		0.64, // the measurement value
		['host' => 'server1', 'location' => 'EU-DE-22'], // optional tags
		['cpucount' => 8], // optional additional fields
);

// write to the database using the path
$result = $client->write(["url" => "api/v1/influxdb/write?db=default"], $point);
var_dump($result);
